src/vm.rs

```diff
@@ -195,11 +199,14 @@ impl From<EcallResult<u64>> for EcallAction {
     }
 }
 
+type AttestationSha384 = AttestationManager<sha2::Sha384>;
+
 /// A VM that is being run.
 pub struct Vm<T: GuestStagePageTable, S = VmStateFinalized> {
     vcpus: VmCpus,
     vm_pages: VmPages<T, S>,
     guests: Option<Guests<T>>,
+    attestation_mgr: Mutex<AttestationSha384>,
```

Comment:
Might it be cleaner to move the mutex inside `AttestationManager`? That way you don't have to do `&mut self.attestation_mgr.lock()` everywhere.